examples/pytorch_lightning/relational_gat.py

```diff
@@ -0,0 +1,146 @@
+from typing import Dict, List, Tuple, Optional
+from torch_geometric.typing import NodeType, EdgeType
+
+import torch
+from torch import Tensor
+import torch.nn.functional as F
+from torchmetrics import Accuracy
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning import LightningDataModule, LightningModule, Trainer
+
+import torch_geometric.transforms as T
+from torch_geometric.data import Batch
+from torch_geometric import seed_everything
+from torch_geometric.datasets import OGB_MAG
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import GATConv, Linear, to_hetero
+
+
+class DataModule(LightningDataModule):
+    def __init__(self, root: str):
+        super().__init__()
+        self.data = OGB_MAG(root, preprocess='metapath2vec',
```

Comment:
This should be done in `prepare_data`, as you don't want multi process to execute it and preprocess the data.